File: 403/python/3.py
N, M, Q = map(int, input().split())

admin = [set() for _ in range(N)]
full = [False] * N 
output = []

for _ in range(Q):
  query = list(map(int, input().split()))
  if query[0] == 1:
    x, y = query[1] - 1, query[2] - 1
    if not full[x]:
      admin[x].add(y)
  elif query[0] == 2:
    x = query[1] - 1
    full[x] = True
    admin[x].clear()
  elif query[0] == 3:
    x, y = query[1] - 1, query[2] - 1
    if full[x] or y in admin[x]:
      output.append("Yes")
    else:
     output.append("No")

print("\n".join(output))

File: 403/3.cpp
#include <iostream>
#include <vector>
#include <unordered_set>
using namespace std;

int main() {
  int N, M, Q;
  cin >> N >> M >> Q;

  vector<bool> full(N, false);
  vector<unordered_set<int>> admin(N);

  for (int i = 0; i < Q; i++){
    int type;
    cin >> type;
    int idx;
    cin >> idx;
    idx--; 

    if(type == 1) {
      int x;
      cin >> x;
      if(!full[idx])
        admin[idx].insert(x);
    }
    else if(type == 2){
      full[idx] = true;
      admin[idx].clear();
    }
    else if(type == 3){
      int x;
      cin >> x;
      if(full[idx] || admin[idx].count(x)) {
        cout << "Yes" << endl;
      } else {
        cout << "No" << endl;
      }
    }
  }
  
  return 0;
}